examples/io_test/io_plane_test.f90

```diff
@@ -54,16 +55,10 @@ program io_plane_test
    call decomp_2d_testing_log()
 
    call decomp_2d_io_init()
-   call decomp_2d_init_io(io_name)
-   call decomp_2d_register_variable(io_name, "x_pencil-x_plane.dat", 1, 0, 1, mytype)
-   call decomp_2d_register_variable(io_name, "x_pencil-y_plane.dat", 1, 0, 2, mytype)
-   call decomp_2d_register_variable(io_name, "x_pencil-z_plane.dat", 1, 0, 3, mytype)
-   call decomp_2d_register_variable(io_name, "y_pencil-x_plane.dat", 2, 0, 1, mytype)
-   call decomp_2d_register_variable(io_name, "y_pencil-y_plane.dat", 2, 0, 2, mytype)
-   call decomp_2d_register_variable(io_name, "y_pencil-z_plane.dat", 2, 0, 3, mytype)
-   call decomp_2d_register_variable(io_name, "z_pencil-x_plane.dat", 3, 0, 1, mytype)
-   call decomp_2d_register_variable(io_name, "z_pencil-y_plane.dat", 3, 0, 2, mytype)
-   call decomp_2d_register_variable(io_name, "z_pencil-z_plane.dat", 3, 0, 3, mytype)
+
+   call decomp_2d_io_register_var2d("x_pencil-x_plane.dat", 1, real_type)
+   call decomp_2d_io_register_var2d("y_pencil-y_plane.dat", 2, real_type)
+   call decomp_2d_io_register_var2d("z_pencil-z_plane.dat", 3, real_type)
```

Comment:
Where do we pass information about coarsening? ADIOS2 needs this during variable registration.

Reply:
My plan is to allow coarsening using a dedicated decomp_info object and the interpolation module. When the variable is registered, the decomp_info object of the coarse grid can be provided.